R/as_xml_document.R

```diff
@@ -63,3 +63,17 @@ as_xml.list <- function(x, ..., missing = "elem") {
   add_node(x, doc)
   xml_root(doc)
 }
+
+#' @export
+as_xml_document.xml_node <- function(x, ...) {
+  xml_new_root(.value = x, ..., .copy = TRUE)
+}
+
+#' @export
+as_xml_document.xml_nodeset <- function(x, root, ...) {
+  doc <- xml_new_root(.value = root, ...)
+  for (i in seq_along(x)) {
+    xml_add_child(doc, x[[i]])
+  }
+  doc
+}
```

Comment:
Probably should add `as_xml_document.xml_document()` for completeness (should just return `x`)